build.sbt

```diff
@@ -1,42 +1,65 @@
-testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+lazy val publishSettings = Def.settings(
+  sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch")),
+  // indicate the open source licenses that apply to our project
+  licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+  // publish to the Sonatype repository
+  publishTo := sonatypePublishToBundle.value,
+  Compile / doc / target := file("site-output"),
+  // binary compatibility check
+  mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
+)
 
-description := "A library that does nothing useful"
+lazy val site = project
+  .in(file("site"))
+  .settings(siteSettings)
+  .settings(publish / skip := true)
+  .enablePlugins(MdocPlugin)
+  .dependsOn(library)
+
+lazy val siteSettings = {
+  /*
+  Combine scaladoc and mdoc.
+  To do so,
+   1. copy site to target/processed_site
+   2. use mdoc to process site/docs/ **.md
+   3. mdoc outputs new md files in target/processed_site/docs
+   4. scaladoc runs and outputs files in site-ouput
+  */
+  val tmpSiteDir = "processed_site"
+  val outputSiteDir = "site-output"
+
+  val copySiteTask = Def.task {
+    IO.copyDirectory(
+      baseDirectory.value / "site",
+      target.value / tmpSiteDir
+    )
+  }
+
+  Def.settings(
+    library / Compile / doc / target := (LocalRootProject / target).value / outputSiteDir,
+    mdocIn :=  baseDirectory.value / "docs",
+    mdocOut := (LocalRootProject / target).value / tmpSiteDir / "docs",
+    mdocVariables := Map("VERSION" -> version.value),
+    Compile / doc := (library / Compile / doc).dependsOn(Def.sequential(copySiteTask, mdoc.toTask(""))).value,
+    library / Compile / doc / scalacOptions ++= Seq(
+      "-siteroot", ((LocalRootProject / target).value / tmpSiteDir).absolutePath,
+    )
+  )
+}
 
-import xerial.sbt.Sonatype._
-sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch"))
-// indicate the open source licenses that apply to our project
-licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt"))
-// publish to the Sonatype repository
-publishTo := sonatypePublishToBundle.value
-
-// documentation website
-enablePlugins(ParadoxPlugin, ParadoxSitePlugin, MdocPlugin, SiteScaladocPlugin)
-mdocIn := sourceDirectory.value / "documentation"
-mdocExtraArguments += "--no-link-hygiene"
-Compile / paradox / sourceDirectory := mdocOut.value
-makeSite := makeSite.dependsOn(mdoc.toTask("")).value
-SiteScaladoc / siteSubdirName := "api"
-paradoxProperties += ("scaladoc.base_url" -> "api")
-
-// binary compatibility check
-mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
-
-inThisBuild(List(
-  semanticdbEnabled := true,
-  semanticdbVersion := scalafixSemanticdb.revision,
-  scalafixScalaBinaryVersion := "2.13",
-
-  scalacOptions ++= Seq(
-    "-Yrangepos",
-    "-Ywarn-unused"
-  ),
-))
```

Comment:
IMHO the complexity of `build.sbt` has increased too much. But maybe it’s OK to keep it like that for now, maybe we will be able to simplify it later, e.g., when we will use the built-in snippets compiler instead of mdoc.

Reply:
I think that we have no choice but to go this way. Even when the snippet compiler will land, it might take some time to reach feature parity with mdoc.